pkgs/tools/typesetting/tex/mftrace/default.nix

```diff
@@ -0,0 +1,51 @@
+{ stdenv, fetchFromGitHub, lib, makeWrapper, autoconf, python3, fontforge, potrace }:
+stdenv.mkDerivation rec {
+  pname = "mftrace";
+  version = "1.2.20";
+
+  # https://lilypond.org/download/sources/mftrace/mftrace-1.2.20.tar.gz
+  # is incomplete, fetch repo and use autoconf instead
+  # see https://github.com/hanwen/mftrace/issues/13
+  src = fetchFromGitHub {
+    owner = "hanwen";
+    repo = "mftrace";
+    rev = "release/${version}";
+    sha256 = "02ik25aczkbi10jrjlnxby3fmixxrwm2k5r4fkfif3bjfym7nqbc";
+  };
+
+  # the potrace/autotrace detection is done at runtime
+  # and can be omitted at build time
+  patches = [ ./disable-build-time-trace-detection.patch ];
+
+  nativeBuildInputs = [ makeWrapper autoconf python3 ];
+
+  buildInputs = extraBuildInputs;
+
+  # at runtime, mftrace needs:
+  # - a tex distribution containing the fonts and the binaries
+  #   kpsewich, mf (metafont), t1asm, mktextfm
+  # - one of potrace (preferred), autotrace
+  # - one of fontforge (preferred), ghostscript
+  extraBuildInputs = [ fontforge potrace ];
+
+  preConfigure = ''
+    autoconf
+  '';
+
+  postInstall = lib.optionalString (extraBuildInputs != [ ]) ''
```

Comment:
This can't be overriden right now.

Reply:
At last, I have read the source of `mkDerivation`, I get it now. That doesn't work as I thought.

Would passing `extraBuildInputs` as an argument be reasonable?

Reply:
Just for the record, I tried implementing a `passthru.withEnv` function that wraps a minimal build (i.e. without `postInstall`) with whatever you pass it. I took inspiration from some `passthru.withPlugins` I have seen around. It worked very well, but it also felt overkill (I had to create an `mftraceMinimal` as well to keep things properly overrideable).

Reply:
@SuperSandro2000 is it ok if I hardcode `potrace` and `fontforge` inside `postInstall`, say after checking that they are not `null`? Then one can write e.g. `fontforge = ghostscript` or `fontforge = null` if they want to override those.

On a related note, I'll try to make make `mftrace` compatible with `texlive.combine`, so if one wants to bind `mftrace` with a particular `texlive` instance they can use that mechanism. It sounds better than what I was trying to do here.

Reply:
I changed the mechanism so that the dependencies can be overridden via e.g. `fontforge = ghostscript` (the options are described in a comment). I have also made it `texlive.combine`-able.

Reply:
> after checking that they are not null

You don't need to check if packages are not null in inputs and overriding packages to null in inputs is not encouraged.

Reply:
Indeed, I noticed that `makeBinPath` works just fine with `null`. The new code does not check for it.